crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -776,4 +779,430 @@ pub trait Call: LoadState + SpawnBlocking {
+    {
+        // we want to disable this in eth_call, since this is common practice used by other node
+        // impls and providers <https://github.com/foundry-rs/foundry/issues/4388>
+        cfg.disable_block_gas_limit = true;
+
+        // Disabled because eth_call is sometimes used with eoa senders
+        // See <https://github.com/paradigmxyz/reth/issues/1959>
+        cfg.disable_eip3607 = true;
+
+        // The basefee should be ignored for eth_call
+        // See:
+        // <https://github.com/ethereum/go-ethereum/blob/ee8e83fa5f6cb261dad2ed0a7bbcde4930c41e6c/internal/ethapi/api.go#L985>
+        cfg.disable_base_fee = true;
+
+        // apply block overrides, we need to apply them first so that they take effect when we we
+        // create the evm env via `build_call_evm_env`, e.g. basefee
+        if let Some(mut block_overrides) = overrides.block {
+            if let Some(block_hashes) = block_overrides.block_hash.take() {
+                // override block hashes
+                db.block_hashes
+                    .extend(block_hashes.into_iter().map(|(num, hash)| (U256::from(num), hash)))
+            }
+            apply_block_overrides(*block_overrides, &mut block);
+        }
+
+        let request_gas = request.gas;
+        let mut env = self.build_call_evm_env(cfg, block, request)?;
+        // set nonce to None so that the next nonce is used when transacting the call
+        env.tx.nonce = None;
+
+        // apply state overrides
+        if let Some(state_overrides) = overrides.state {
+            apply_state_overrides(state_overrides, db)?;
+        }
+
+        if request_gas.is_none() {
+            // No gas limit was provided in the request, so we need to cap the transaction gas limit
+            if env.tx.gas_price > U256::ZERO {
+                // If gas price is specified, cap transaction gas limit with caller allowance
+                trace!(target: "rpc::eth::call", ?env, "Applying gas limit cap with caller allowance");
+                cap_tx_gas_limit_with_caller_allowance(db, &mut env.tx)?;
+            } else {
+                // If no gas price is specified, use maximum allowed gas limit. The reason for this
+                // is that both Erigon and Geth use pre-configured gas cap even if
+                // it's possible to derive the gas limit from the block:
+                // <https://github.com/ledgerwatch/erigon/blob/eae2d9a79cb70dbe30b3a6b79c436872e4605458/cmd/rpcdaemon/commands/trace_adhoc.go#L956
+                // https://github.com/ledgerwatch/erigon/blob/eae2d9a79cb70dbe30b3a6b79c436872e4605458/eth/ethconfig/config.go#L94>
+                trace!(target: "rpc::eth::call", ?env, "Applying gas limit cap as the maximum gas limit");
+                env.tx.gas_limit = gas_limit;
+            }
+        }
+
+        Ok(env)
+    }
+}
+
+/// Helper type for representing the fees of a [`TransactionRequest`]
+#[derive(Debug)]
+pub struct CallFees {
+    /// EIP-1559 priority fee
```

Comment:
If I move CallFees to crates/rpc/rpc-eth-types/src/revm_utils.rs. I need to Change CallFees members all public. But I don't see CallFees in any other place. Is it Ok? @emhane